src/enforcer.js

```diff
@@ -158,6 +124,73 @@ export class TrustedTypesEnforcer {
      * @private {Object<string, function(*): *|undefined>}
      */
     this.originalSetters_ = {};
+    /**
+     * The object that will be monkey patched by the polyfill.
+     * @private {Window}
+     */
+    this.windowObject = windowObject ||
+      (typeof window !== 'undefined' ? window : null);
+
+    if (!this.windowObject || typeof this.windowObject !== 'object') {
+      throw new Error(
+          // eslint-disable-next-line
+          'The polyfill expects a global `window` object or emulated `window-like` object passed to the enforcer as second argument'
+      );
+    }
+
+    // In IE 11, insertAdjacent(HTML|Text) is on HTMLElement prototype
+    this.insertAdjacentObjectPrototype = ((w) => {
+      if (!w.Element || !w.HTMLElement) return null;
+      return apply(hasOwnProperty, w.Element.prototype,
+          ['insertAdjacentHTML']) ?
+          w.Element.prototype :
+            w.HTMLElement.prototype;
+    })(this.windowObject);
+  }
+
+  /**
+     * Checks whether the value is instanceOf the specific window object.
+     * @param {*} value
+     * @param {!string} winProp
+     * @return {!boolean}
+     * @private
+     */
+  instanceOfDomProperty(value, winProp) {
+    const obj = this.windowObject[winProp];
+
+    return value instanceof obj;
+  }
+
+  /**
+ * Converts an uppercase tag name to an element constructor function name.
+ * Used for property setter hijacking only.
+ * @param {string} tagName
+ * @return {string|null}
+ */
+  convertTagToConstructor(tagName) {
+    if (tagName == '*') {
+      return 'HTMLElement';
+    } else if (!this.windowObject.document) {
+      return null;
+    } else {
+      return this.getConstructorName_(
+          this.windowObject.document.createElement(tagName).constructor
+      );
+    }
+  }
+
+  /**
+ * Return object constructor name
+ * (their function.name is not available in IE 11).
+ * @param {Function} fn
+ * @return {string|null}
+ * @private
+ */
+  getConstructorName_(fn) {
+    if (!this.windowObject.document) return null;
+    return this.windowObject.document.createElement('div').constructor.name ?
+      fn.name :
+      ('' + fn).match(/^\[object (\S+)\]$/)[1];
   }
 
   /**
```

Comment:
Can you mention in install() description that now the unknown properties are not hooked, instead of throwing, and why this was needed?

Reply:
Done.